src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,181 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+
+import java.util.InputMismatchException;
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+
+import task.Task;
+
+import exception.InvalidCommandException;
+import exception.MissingInfoException;
+
+/**
+ * Parser deals with making sense of the user command.
+ */
+public class Parser {
+
+    private Scanner input;
+
+    public Parser(Scanner input) {
+        this.input = input;
+    }
+
+    /**
+     * Sets the scanner input stored in this object.
+     *
+     * @param scanner The scanner to store.
+     */
+    public void setScanner(Scanner scanner) {
+        this.input = scanner;
+    }
+
+    /**
+     * Takes in the command entered by the user and execute it accordingly.
+     *
+     * @param taskList All the tasks in the to-do list.
+     * @return String The relevant reply according to command.
+     */
+    public String executeCommand(TaskList taskList) {
+        String command = this.input.next();
+
+        if (command.equals("bye")) {
```

Comment:
I like the readability of the if-else statements here. It's easy to look for a specific command and see the code for each one :)